tendrl/gluster_integration/sds_sync/vol_utilization.py

```diff
@@ -18,42 +19,55 @@ def computeVolumeStats(data):
             'sizeUsed': float(used)}
 
 
-def showVolumeUtilization(vname):
-    try:
-        data = gfapi.getVolumeStatvfs(vname)
-    except gfapi.GlusterLibgfapiException:
-        sys.stderr.write("CRITICAL: Failed to get the "
-                         "Volume Utilization Data\n")
-        sys.exit(-1)
-    volumeCapacity = computeVolumeStats(data)
-    # total size in KB
-    total_size = volumeCapacity['sizeTotal'] / BYTES_IN_KB
-    # Available free size in KB
-    free_size = volumeCapacity['sizeFree'] / BYTES_IN_KB
-    # used size in KB
-    used_size = volumeCapacity['sizeUsed'] / BYTES_IN_KB
-    vol_utilization = (used_size / total_size) * 100
-    used_inode = data.f_files - data.f_ffree
-    total_inode = data.f_files
-    pcnt_inode_used = (float(used_inode) / total_inode) * 100
-    return (
-        {
-            'total': total_size,
-            'free': free_size,
-            'used': used_size,
-            'pcnt_used': vol_utilization,
-            'total_inode': total_inode,
-            'used_inode': used_inode,
-            'pcnt_inode_used': pcnt_inode_used
-        }
-    )
+def showVolumeUtilization(vnames):
+    volumes = vnames.split(',')
+    ret_dict = {}
+    for volume in volumes:
+        if volume != "":
+            try:
+                data = gfapi.getVolumeStatvfs(volume)
+            except gfapi.GlusterLibgfapiException:
+                sys.stderr.write("CRITICAL: Failed to get the "
+                                 "Volume Utilization Data\n")
+                sys.exit(-1)
```

Comment:
I thought we can continue with other volumes if anyone volume is failed here, but i feel if we remove sys.exit(-1) again it will lead to memory leak problem. so this approach is fine.

Reply:
Ack. Next time anyway it will try to get details again.